tests/components/application-connector/README.md

```diff
@@ -2,65 +2,172 @@
 
-### Local execution
+### Deploy Kyma cluster locally
 
 1. Provision a local Kubernetes cluster with k3d:
-
-   ```shell
+   ```sh
    kyma provision k3d
    ```
 
-2. Install the minimal set of components required to run Application Gateway for either Kyma OS or SKR:
+1. Install the minimal set of components required to run Application Gateway for either Kyma OS or SKR:
+
+    <div tabs name="Kyma flavor" group="minimal-kyma-installation">
+    <details open>
+    <summary label="OS">
+    Kyma OS
+    </summary>
+
+    ```sh
+    kyma deploy --components-file ./resources/installation-config/mini-kyma-os.yaml
+    ```
+
+    </details>
+    <details>
+    <summary label="SKR">
+    SKR
+    </summary>
+
+    ```bash
+    kyma deploy --components-file ./resources/installation-config/mini-kyma-skr.yaml 
+    ```
+
+    </details>
+    </div>
+
+    >**TIP:** More on Kyma installation can be found in [official Kyma documentation](https://kyma-project.io/docs/kyma/latest/02-get-started/01-quick-install/#install-kyma)
+
+### Run the tests
+
+``` sh
+make test-gateway
+```
+
+By default tests clean up after themselves, removing all created resources
+and `test` Namespace.
+
+> **CAUTION:** This might override or remove existing resources, 
+> if names of already existing resources collide with names used by tests
+
+## Debugging
 
-   <div tabs name="Kyma flavor" group="minimal-kyma-installation">
-      <details open>
-      <summary label="OS">
-      Kyma OS
-      </summary>
+### Running locally
 
-   ```bash
-   kyma deploy --components-file ./resources/installation-config/mini-kyma-os.yaml
+The Test Job must run on a cluster, because of the way it accesses Application CRs.
+Application Gateway and Mock Application can both be run locally.
+
+To run Mock Application locally:
+1. Change all `targetUrl` in [Application CRs](./resources/charts/gateway-test/templates/applications/)
+   to reflect new app URL. For example `http://localhost:8081/v1/api/unsecure/ok`
+1. Change all `centralGatewayUrl` to reflect new Application Gateway URL. 
+   For example `http://localhost:8080/positive-authorisation/unsecure-always-ok`
+1. Deploy all resources (you can omit test Job and Central Gateway, but it's easier to just let them fail)
+   on the cluster
+1. Build the Mock Application:
+   
+   <div tabs name="Mock App Build Flavor" group="mock-app-flavor">
+   <details open>
+   <summary label="dockerized">
+   Docker
+   </summary>
+
+   ```shell
+   export DOCKER_TAG="local"
+   export DOCKER_PUSH_REPOSITORY="{Docker username}"
+   make image-mock-app
    ```
-      </details>
-      <details>
-      <summary label="SKR">
-      SKR
-      </summary>
-
-   ```bash
-   kyma deploy --components-file ./resources/installation-config/mini-kyma-skr.yaml 
+
+   </details>
+   <details>
+   <summary label="local">
+   Local
+   </summary>
+
+   Change hardcoded application port in [config.go](./tools/external-api-mock-app/config.go), then
```

Comment:
```suggestion
   Change the hardcoded application port in [`config.go`](./tools/external-api-mock-app/config.go), and run:
```
- Use articles.
- Format file names as `code`. See the [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#use-code-font-for-these-items) for reference.
- Don't break a sentence half-way.